cmd/podman/pull.go

```diff
@@ -122,68 +123,81 @@ func pullCmd(c *cliconfig.PullValues) (retError error) {
 		dockerRegistryOptions.DockerInsecureSkipTLSVerify = types.NewOptionalBool(!c.TlsVerify)
 	}
 
-	// Possible for docker-archive to have multiple tags, so use LoadFromArchiveReference instead
-	if strings.HasPrefix(img, dockerarchive.Transport.Name()+":") {
-		srcRef, err := alltransports.ParseImageName(img)
+	// Special-case for docker-archive which allows multiple tags.
+	if strings.HasPrefix(imgArg, dockerarchive.Transport.Name()+":") {
+		srcRef, err := alltransports.ParseImageName(imgArg)
 		if err != nil {
-			return errors.Wrapf(err, "error parsing %q", img)
+			return errors.Wrapf(err, "error parsing %q", imgArg)
 		}
 		newImage, err := runtime.LoadFromArchiveReference(getContext(), srcRef, c.SignaturePolicy, writer)
 		if err != nil {
-			return errors.Wrapf(err, "error pulling image from %q", img)
+			return errors.Wrapf(err, "error pulling image from %q", imgArg)
 		}
 		fmt.Println(newImage[0].ID())
-	} else {
-		authfile := getAuthFile(c.String("authfile"))
-		spec := img
-		systemContext := image.GetSystemContext("", authfile, false)
-		srcRef, err := alltransports.ParseImageName(spec)
+
+		return nil
+	}
+
+	authfile := getAuthFile(c.String("authfile"))
+
+	if !c.Bool("all-tags") {
+		newImage, err := runtime.New(getContext(), imgArg, c.SignaturePolicy, authfile, writer, &dockerRegistryOptions, image.SigningOptions{}, true, nil)
 		if err != nil {
-			dockerTransport := "docker://"
-			logrus.Debugf("error parsing image name %q, trying with transport %q: %v", spec, dockerTransport, err)
-			spec = dockerTransport + spec
-			srcRef2, err2 := alltransports.ParseImageName(spec)
-			if err2 != nil {
-				return errors.Wrapf(err2, "error parsing image name %q", img)
-			}
-			srcRef = srcRef2
-		}
-		var names []string
-		if c.Bool("all-tags") {
-			if srcRef.DockerReference() == nil {
-				return errors.New("Non-docker transport is currently not supported")
-			}
-			tags, err := docker.GetRepositoryTags(ctx, systemContext, srcRef)
-			if err != nil {
-				return errors.Wrapf(err, "error getting repository tags")
-			}
-			for _, tag := range tags {
-				name := spec + ":" + tag
-				names = append(names, name)
-			}
-		} else {
-			names = append(names, spec)
+			return errors.Wrapf(err, "error pulling image %q", imgArg)
 		}
-		var foundIDs []string
-		foundImage := true
-		for _, name := range names {
-			newImage, err := runtime.New(getContext(), name, c.String("signature-policy"), authfile, writer, &dockerRegistryOptions, image.SigningOptions{}, true, nil)
-			if err != nil {
-				logrus.Errorf("error pulling image %q", name)
-				foundImage = false
-				continue
-			}
-			foundIDs = append(foundIDs, newImage.ID())
-		}
-		if len(names) == 1 && !foundImage {
-			return errors.Wrapf(err, "error pulling image %q", img)
-		}
-		if len(names) > 1 {
-			fmt.Println("Pulled Images:")
+		imgID := newImage.ID()
+		fmt.Println(imgID)
```

Comment:
(Absolutely non-blocking: I wouldn’t bother with the `imgID` variable.)

Reply:
Good catch, fixed.